src/Storages/ObjectStorage/DataLakes/Iceberg/PartitionPruning.cpp

```diff
@@ -1,130 +1,135 @@
+#include "config.h"
+
+#if USE_AVRO
+
 #include <Columns/ColumnNullable.h>
 #include <Columns/ColumnsDateTime.h>
 #include <Common/DateLUTImpl.h>
 #include <DataTypes/DataTypeNullable.h>
+#include <Common/logger_useful.h>
+#include <Parsers/ASTFunction.h>
+#include <Parsers/ASTIdentifier.h>
+#include <Parsers/ASTExpressionList.h>
+#include <Parsers/ASTLiteral.h>
+#include <IO/ReadHelpers.h>
+#include <Common/quoteString.h>
 
 #include <Storages/ObjectStorage/DataLakes/Iceberg/PartitionPruning.h>
-
-namespace DB::ErrorCodes
-{
-extern const int BAD_ARGUMENTS;
-}
+#include <Storages/ObjectStorage/DataLakes/Iceberg/ManifestFile.h>
 
 using namespace DB;
 
 namespace Iceberg
 {
 
-Iceberg::PartitionTransform getTransform(const String & transform_name)
+DB::ASTPtr getASTFromTransform(const String & transform_name, const String & column_name)
 {
     if (transform_name == "year" || transform_name == "years")
     {
-        return Iceberg::PartitionTransform::Year;
+        return makeASTFunction("toYearNumSinceEpoch", std::make_shared<DB::ASTIdentifier>(column_name));
     }
     else if (transform_name == "month" || transform_name == "months")
     {
-        return Iceberg::PartitionTransform::Month;
+        return makeASTFunction("toMonthNumSinceEpoch", std::make_shared<DB::ASTIdentifier>(column_name));
     }
     else if (transform_name == "day" || transform_name == "date" || transform_name == "days" || transform_name == "dates")
     {
-        return Iceberg::PartitionTransform::Day;
+        return makeASTFunction("toRelativeDayNum", std::make_shared<DB::ASTIdentifier>(column_name));
     }
     else if (transform_name == "hour" || transform_name == "hours")
     {
-        return Iceberg::PartitionTransform::Hour;
+        return makeASTFunction("toRelativeHourNum", std::make_shared<DB::ASTIdentifier>(column_name));
     }
     else if (transform_name == "identity")
     {
-        return Iceberg::PartitionTransform::Identity;
+        return std::make_shared<ASTIdentifier>(column_name);
+    }
+    else if (transform_name.starts_with("truncate"))
+    {
+        auto argument_start = transform_name.find('[');
```

Comment:
Is `transform_name` validated somewhere else and are we guaranteed to always have the correct format here? Should we be also be checking if there's a `transform_name.find(']')` and use that while calculating the width?